tests/protocol/test_wire.py

```diff
@@ -746,110 +743,115 @@ def test_fetch_manifest_should_update_goal_state_and_not_change_default_channel_
-                        with patch("azurelinuxagent.common.protocol.wire.WireClient.get_artifacts_profile_through_host",
-                                   wraps=protocol.client.get_artifacts_profile_through_host) as patch_host:
-                            ret = protocol.client.get_artifacts_profile()
-                            self.assertIsInstance(ret, InVMArtifactsProfile)
-
-                            self.assertEquals(patch_host.call_count, 1)
-                            # The host channel calls the direct function under the covers
-                            self.assertEquals(patch_direct.call_count, 1 + patch_host.call_count)
-                            self.assertEquals(mock_update_goal_state.call_count, 0)
-
-                            self.assertEquals(HostPluginProtocol.is_default_channel(), True)
-
-    @patch("azurelinuxagent.common.protocol.wire.WireClient.get_goal_state")
-    @patch("azurelinuxagent.common.protocol.hostplugin.HostPluginProtocol.get_artifact_request")
-    def test_get_artifacts_profile_should_retry_the_host_channel_after_refreshing_the_host_plugin(self,
-                                                                                                  mock_get_artifact_request,
-                                                                                                  *args):
-        mock_get_artifact_request.return_value = "dummy_url", "dummy_header"
-        mock_failed_response = MockResponse(body=b"", status_code=httpclient.GONE)
-        mock_successful_response = MockResponse(body=b'{ "onHold": true }', status_code=200)
-
-        HostPluginProtocol.set_default_channel(False)
-
-        # Direct channel fails, host channel fails due to stale goal state, host channel succeeds after refresh.
-        # As a consequence, host plugin should have been updated and host channel should have been set as default.
-        with create_mock_protocol(artifacts_profile_blob="testurl") as protocol:
-            with patch("azurelinuxagent.common.utils.restutil._http_request",
-                       side_effect=[mock_failed_response, mock_failed_response, mock_successful_response]):
-                with patch("azurelinuxagent.common.protocol.wire.WireClient.update_host_plugin_from_goal_state") as mock_update_host_plugin_from_goal_state:
-                    with patch("azurelinuxagent.common.protocol.wire.WireClient.fetch", wraps=protocol.client.fetch) as patch_direct:
-                        with patch("azurelinuxagent.common.protocol.wire.WireClient.get_artifacts_profile_through_host",
-                                   wraps=protocol.client.get_artifacts_profile_through_host) as patch_host:
-                            ret = protocol.client.get_artifacts_profile()
-                            self.assertIsInstance(ret, InVMArtifactsProfile)
+    def test_get_artifacts_profile_should_not_invoke_host_channel_when_direct_channel_succeeds(self):
```

Comment:
Several tests for the artifacts profile blob broke due to the refactoring. Rewrote them. There is a lot of common setup in many of the tests in this file; I will have to rewrite many of them as I continue refactoring the protocol; will do that on subsequent PRs